internal/cloud/azure/application_identity.go

```diff
@@ -45,38 +46,53 @@ func (c *AzureConfig) NewServicePrincipalsClient(ctx context.Context) (ServicePr
 }
 
 type ApplicationIdentityConfig struct {
-	ID            string
-	Name          string
-	ApplicationID string
-	ClientID      string
-	ClientSecret  string
+	ID                     string
+	Name                   string
+	ApplicationID          string
+	ServicePrincipalID     string
```

Comment:
ClientID is an overloaded term. Azure applications can have a `client_id` (different than the actual ID 👍 ), but permissions can't be attached to applications.  Only to service principles, which are "owned" by applications (but somehow unviewable in the console???).

Its messy, but I changed the name here to be explicit that this is a **Service Principal** ID, not the application `client_id`.

Reply:
How do you access the id and secret from the terraform resource?

Reply:
The resource `id` field will be the Service Principal ID, but both fields are directly accessible in the azure-specific output block.  You can see I updated the name of them here: https://github.com/massdriver-cloud/terraform-provider-mdxc/pull/25/files#diff-08d112e976e2479674666bd6acea8430fd217225c39ce5186b3e1a4788d52ad1L86-L92